lib/internal/repl.js

```diff
@@ -164,13 +164,22 @@ function setupHistory(repl, historyPath, oldHistoryPath, ready) {
       }
     }
 
-    fs.open(historyPath, 'w', onhandle);
+    fs.open(historyPath, 'r+', onhandle);
   }
 
   function onhandle(err, hnd) {
     if (err) {
       return ready(err);
     }
+    fs.ftruncate(hnd, 0, (err) => {
```

Comment:
Previous implementation cleared the file once when opening it with `w`. This was the only place that would reset history file content - i.e. if one would set `NODE_REPL_HISTORY_SIZE` to value lower than 1000, then the history file would be trimmed when node starts. This `ftruncate` call here is to preserve this functionality.